src/bootstrap/bootstrap.py

```diff
@@ -250,22 +246,15 @@ def default_build_triple(verbose):
             if verbose:
                 print("pre-installed rustc not detected: {}".format(e),
                     file=sys.stderr)
-                print("falling back to auto-detect", file=sys.stderr)
-
-    required = not platform_is_win32()
-    ostype = require(["uname", "-s"], exit=required)
-    cputype = require(['uname', '-m'], exit=required)
+                print("assuming x86_64-pc-windows-msvc", file=sys.stderr)
+            return 'x86_64-pc-windows-msvc'
 
-    # If we do not have `uname`, assume Windows.
-    if ostype is None or cputype is None:
-        return 'x86_64-pc-windows-msvc'
-
-    ostype = ostype.decode(default_encoding)
-    cputype = cputype.decode(default_encoding)
+    uname = require(["uname", "-smop"]).decode(default_encoding)
```

Comment:
Save a few cycles time by lifting all the `uname` invocations to a single command